learner/sampler/slurper.py

```diff
@@ -0,0 +1,134 @@
+from itertools import islice
+import math
+import os
+import random
+import sys
+import time
+from typing import List
+
+##TO DO:
+#Only call this when the file is of more than ~1/3 total available memory size of the computer. (or approaching max of python 2)
+#Ensure this works for unstructured file types - write this for xml and json
+#Catch encoding errors inside the functions (get filesize)
+#Connect to the API for the rest of the project
+
+class Slurper(object):
+    def __init__(self, filename: str, encoding: str = 'UTF-8') -> None:
+        self.file = filename
+        self.encoding = check_file_encodings()
+        self.nlines = estimate_line_size()
+        self.sample_size = get_sample_size()
+    
+    def check_file_encodings(file: str = None, encoding: str = None) -> str:
+        """Ensure that the file can be opened by cycling through common file types."""        
+        if not file:
+            file = self.filename
+        if not encoding:
+            encoding = self.encoding            
+
+        try:
+            with open(file, encoding = encoding) as f:
+                f.seek(1000,0)
+                f.readline()
+                f.close()
+            return encoding
+        except:
+            common_encodings = ["Latin-1", "UTF-16", "ascii", "cp037", "cp437", "UTF-32"]
```

Comment:
should be a state variable, not a local variable. "magic values" = bad paradigm.